### PR TITLE
BigInteger support 

### PR DESCRIPTION
**Pull request by kartik-gupta-ij**
#6
Fixed using [json-bigint](https://www.npmjs.com/package/json-bigint)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -98,6 +99,15 @@
...
   },
+  "pnpm": {
+    "overrides": {
+      "@babel/traverse": "7.23.2",
+      "semver": "7.5.2",
+      "tough-cookie": "4.1.3",
+      "word-wrap": "1.2.4",
+      "graphql": " 16.8.1 "
+    }
+  },
```
Comment (generall) [inline package.json:109]: why this is changed?

**Comment by generall** [comment]
shouldn't this be also changed https://github.com/qdrant/openapi-typescript-fetch/blob/f33b84870048e75efe68e1ebaf6876bf87548ab1/src/fetcher.ts#L97C2-L97C79 ?

File: package.json (hunk 1 above)
Comment (kartik-gupta-ij) [reply package.json:109]: Those packages has vulnerabilities

**Comment by kartik-gupta-ij** [comment]
> shouldn't this be also changed https://github.com/qdrant/openapi-typescript-fetch/blob/f33b84870048e75efe68e1ebaf6876bf87548ab1/src/fetcher.ts#L97C2-L97C79 ?

Fixed

**Review by generall — APPROVED** [review]

File: src/fetcher.ts [hunk 2]
```diff
@@ -14,6 +14,7 @@ import {
   _TypedFetch,
   TypedFetch,
 } from './types.js'
+import JSONbig from 'json-bigint'
```
Comment (Rendez) [inline src/fetcher.ts:18]: I think we should default to using BigInt, as [all modern browsers have now native support for it](https://caniuse.com/bigint). That way we tell the lib to internally avoid using `bignumber.js`:

`const JSONbigNative = JSONbig({ useNativeBigInt: true });`

File: package.json (hunk 1 above)
Comment (Rendez) [reply package.json:109]: I would encourage you to explore more modern alternatives. This one was last published 4 years ago, and it seems to have quite a few [issues opened](https://github.com/sidorares/json-bigint/issues).

I found one that might be a better candidate: https://github.com/Ivan-Korolenko/json-with-bigint/blob/main/json-with-bigint.js (much, much smaller and simpler and perhaps even faster?) from this S.O. discussion: https://stackoverflow.com/a/69644630

Comment (kartik-gupta-ij) [reply package.json:109]: @Rendez, The vulnerabilities exist in `jest`, `msw`, `@typescript-eslint`, `jest-environment-jsdom`, and `@changesets/cli`, and they have not been resolved yet.

Comment (Rendez) [reply package.json:109]: It's not just the vulnerabilities, you're adding a massive outdated and unmaintained library, that is my concern.

Comment (Rendez) [reply package.json:109]: > @Rendez, The vulnerabilities exist in `jest`, `msw`, `@typescript-eslint`, `jest-environment-jsdom`, and `@changesets/cli`, and they have not been resolved yet.

Can't we update these libraries to latest? Do they still have these vulnerabilites after updating?

File: tsconfig.base.json [hunk 3]
```diff
@@ -5,7 +5,7 @@
         "declarationDir": "dist",
         "esModuleInterop": true,
         "forceConsistentCasingInFileNames": true,
-        "lib": ["dom", "dom.iterable", "es2019"],
+        "lib": ["dom", "dom.iterable", "es2019", "esnext"],
```
Comment (Rendez) [inline tsconfig.base.json]: Please follow this convention: https://github.com/microsoft/TypeScript/wiki/Node-Target-Mapping

File: src/jsonParser.ts [hunk 4]
```diff
@@ -0,0 +1,41 @@
+export type Json =
```
Comment (Rendez) [inline src/jsonParser.ts]: Why did you add:
```json
{
 "dependencies": {
    "json-with-bigint": "^2.1.0"
  }
}
```

And then copied the [source](https://github.com/Ivan-Korolenko/json-with-bigint/blob/main/json-with-bigint.js) here?

File: test/fetch.test.ts [hunk 5]
```diff
@@ -112,14 +112,14 @@ describe('fetch', () => {
   })
 
   it(`POST /accepted`, async () => {
```
Comment (Rendez) [inline test/fetch.test.ts:115]: We should test the conversion and parsing of responses with big int.

**Review by Rendez — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (kartik-gupta-ij) [reply package.json:109]: Yes, they have vulnerabilities in the latest version too.

File: src/jsonParser.ts (hunk 4 above)
Comment (Rendez) [reply src/jsonParser.ts]: I much rather have the dependency in case new versions with fixes happen, etc.

**Review by Rendez — APPROVED** [review]

**Pull request merged**